wp-content/themes/mstoday/functions.php

```diff
@@ -446,4 +446,67 @@ function mstoday_dequeue_unused_google_inline_sheets_assets() {
 	}
 
 }
-add_filter( 'wp_enqueue_scripts', 'mstoday_dequeue_unused_google_inline_sheets_assets', 9999 );
\ No newline at end of file
+add_filter( 'wp_enqueue_scripts', 'mstoday_dequeue_unused_google_inline_sheets_assets', 9999 );
+
+/**
+ * Conditionally enqueue all of the required JS and CSS for the SiteOrigin page builder plugin + its addons
+ * in order to help with pagespeed and performance. Now these assets should only be loaded on posts or pages
+ * where `siteorigin_panels_render( $post->ID )` isn't empty.
+ * 
+ * @see https://github.com/INN/umbrella-mstoday/issues/95
+ * @see https://siteorigin.com/docs/widgets-bundle/form-building/builder-field/
+ * @see https://siteorigin.com/thread/check-if-page-is-using-the-page-builder/
+ */
+function mstoday_dequeue_unused_site_origin_assets() {
+
+	global $post;
+
+	$maybe_load_assets = false;
+
+	if ( ( is_single() || is_page() ) && siteorigin_panels_render( $post->ID ) ){
+		$maybe_load_assets = true;
+		return;
+	}
+
+	$scripts = array(
+		'siteorigin-premium-web-font-importer',
+		'sow-slider-slider-cycle2',
+		'sow-slider-slider-cycle2-swipe',
+		'sow-google-map',
+		'sow-slider-slider',
+		'sowb-fittext',
+		'sow-cta-main'
+	);
+
+	$styles = array(
+		'sow-slider-slider-cycle2',
+		'sow-slider-slider-cycle2-swipe',
+		'sow-google-map',
+		'sow-slider-slider',
+		'sow-button-base',
+		'sow-cta-main'
+	);
+
+	// loop through /uploads/siteorigin-widgets/* and remove all of those css files
+	// because for some reason the SO widgets bundle plugin creates these files based off of widget settigns
+	$uploads_dir = wp_upload_dir();
+	$siteorigin_uploads = $uploads_dir['basedir'] . '/siteorigin-widgets';
+	$uploads_items = list_files( $siteorigin_uploads, 1 );
+	if( ! empty( $uploads_items ) ) {
+		foreach ( $uploads_items as $item ) {
+			$styles[] = str_replace( '.css', '', basename( $item ) );
+		}
+	}
```

Comment:
That's fair. I left it here since any page that it _shouldn't_ load on will never get to this point since they'll have exited this function above, but I agree it should be moved just to keep everything consistent and easier to maintain. I'll update it now.

Reply:
@benlk All of your edits have been added in https://github.com/INN/umbrella-mstoday/pull/100/commits/f51f2d84008265c7caf18bb3112d5fc834285787. I also updated the Inline Google plugin function to check `is_a( $post, 'WP_Post' )` too